docs/01_key-concepts/05_stake.md

```diff
@@ -1,6 +1,28 @@
 ---
-content_title: Staking on EOSIO based blockchains
-link_text: Staking on EOSIO based blockchains
+content_title: Staking on EOSIO-based blockchains
+link_text: Staking on EOSIO-based blockchains
 ---
 
-On EOSIO based blockchains, to be able to deploy and then interact with a smart contract via its implemented actions it needs to be backed up by resources allocated on the account where the smart contract is deployed to. The three resource types an EOSIO smart contract developer needs to know about are RAM, CPU and NET. You can __stake__ CPU and NET and you can __buy__ RAM. You will also find that staking/unstaking is at times referred to as delegating/undelegating. The economics of staking is also to provably commit to a promise that you'll hold the staked tokens, either for NET or CPU, for a pre-established period of time, in spite of inflation caused by minting new tokens in order to reward BPs for their services every 24 hours.
+## System Resources
+
+EOSIO-based blockchains work with three system resources:
+
+* [RAM](02_ram.md),
+* [CPU](03_cpu.md) and
+* [NET](04_net.md).
+
+## How To Allocate System Resources
+
+Smart contracts deployed on EOSIO-based blockchains need sufficient system resources, RAM, CPU and NET, for the blockchain to be able to execute their code.
+
+### Stake NET and CPU
+
+The CPU and NET system resources are allocated by the smart contract developer via the staking mechanism on the account where the smart contract is deployed to, refer to the [cleos manual](https://developers.eos.io/manuals/eos/v2.0/cleos/how-to-guides/how-to-stake-resource) on how to do it via the command line interface.
+
+You will also find that staking/unstaking is at times referred to as delegating/undelegating. The economics of staking is also to provably commit to a promise that you'll hold the staked tokens, either for NET or CPU, for a pre-established period of time, in spite of inflation caused by minting new tokens in order to reward BPs for their services every 24 hours.
+
+When you stake tokens for CPU and NET, you gain access to system resources proportional to the total amount of tokens staked by all other users for the same system resource at the same time. This means you can perform transactions at no cost but in the limits of the staked tokens. The staked tokens guarantee the proportional amount of resources regardless of any variations in the free market. If a smart contract consumes all its allocated CPU and NET, it needs to wait 24 hours when the system will replenish them or allocate more resources.
```

Comment:
can you explain here how resources are related to the blocks? so we have a block every half a second and we have this 24 hours limit. is staked CPU/NET is amount that one can use during the 24 hours?
I believe there should be limit per block.

Reply:
I think you want to see what is written here right?
https://developers.eos.io/welcome/latest/protocol/consensus_protocol#52-block-production

I am not aware of a relation between blocks and NET.
instead there is a relation of how long a transaction can take in CPU units. https://developers.eos.io/manuals/eosio.cdt/latest/structeosio_1_1blockchain__parameters#variable-min_transaction_cpu_usage
I think it is a good idea to talk about that here as well.

Reply:
I don't see this information in block production page. So what I would like to see is more details on how resource is spent. you mention that resource can be drained and it takes 24hr to restore. so how this process happening? Let's say you staked 10 microseconds of CPU. What does it mean? Is this amount you can spend per block or per 24hr?

Reply:
I don't understand why this statement is not clear
`If a smart contract consumes all its allocated CPU and NET, it needs to wait 24 hours when the system will replenish them or allocate more resources.`

Reply:
take into account that people read this because of they don't understand what this is. Maybe they know only how fiat currency works. So this sounds very unusual.
Maybe it worth to give some example? let's say someone staked some CPU. What is it's unit of measurement? Earlier you mention that it is in microsecond and later that with CPU you gain access to resource proportionally to total amount of resource. So what exactly you have? Absolute time span per 24 hr or the fraction of all available resources? Does it fluctuate over time or is it absolute value?
Next. How one can spend resource? What if one purchased 200 millisecond of CPU? Can they use majority of block resource? I think it worth to mention that limits per block at least exist and in later paragraph user can read more about them. I think It adds up to conceptual understanding of what is CPU.